### ISSUE TITLE
Can not create new certificat

### ISSUE THREAD

**Issue by joselito11** [hassio-addons/addon-nginx-proxy-manager#76]
I have several times deleted, restarted HA, installed addon, but get always the same error. This has happen before I tried with reinstalling the addon and continiue.
# Problem/Motivation

[2/29/2020] [10:33:05 AM] [Nginx    ] › ℹ  info      Reloading Nginx
[2/29/2020] [10:33:05 AM] [SSL      ] › ℹ  info      Requesting Let'sEncrypt certificates for Cert #5: aaa.duckdns.org
[2/29/2020] [10:33:29 AM] [Nginx    ] › ℹ  info      Reloading Nginx
[2/29/2020] [10:33:29 AM] [Express  ] › ⚠  warning   Command failed: /usr/bin/certbot certonly --cert-name "npm-5" --agree-tos --email "xxxx@gmail.com" --preferred-challenges "dns,http" -n -a webroot -d "aaa.duckdns.org" 
Saving debug log to /data/logs/letsencrypt/letsencrypt.log
Plugins selected: Authenticator webroot, Installer None
Obtaining a new certificate
Performing the following challenges:
http-01 challenge for aaa.duckdns.org
Using the webroot path /data/letsencrypt-acme-challenge for all unmatched domains.
Waiting for verification...
Challenge failed for domain aaa.duckdns.org
http-01 challenge for aaa.duckdns.org
Cleaning up challenges
Some challenges have failed

## Expected behavior

create new ssl certificat

## Actual behavior

get internal error

## Steps to reproduce

install addon, create new ssl certificat

## Proposed changes

> (If you have a proposed change, workaround or fix,
> describe the rationale behind it)

**Comment by frenck** [comment]
Make sure:
- The domain resolves to your IP address
- Port 80 from the outside is forwarded to port 80 of the add-on
- Port 443 from the outside is forwarded to port 443 of the add-on

**Comment by TWhidden** [comment]
What do we do, if our ISP (Cox - large internet provider in the USA) blocks port 80. 

Reference:  https://www.cox.com/residential/support/internet-ports-blocked-or-restricted-by-cox.html

On another docker container of mine, I use the letsencrypt dns challenge to get a wildcard certificate.  If needed, I suspect I could bring this over as an alternative, but honestly, I am trying to just use HA for everything.  

Do you think you think the DNS-01 challenge could be added for those of us that have no ability to get tcp80 open?  

https://letsencrypt.org/docs/allow-port-80/ has some alternatives, even though they suggest port 80.

**Comment by MapGuy11** [comment]
I asked for DNS challenge too! When getting a cert On my own it doesn't import either. Because my ISP blocks Open VPN port, 443,80, and 8080

**Comment by flipepiloto** [comment]
I had the same issue

1 - Create on my router a port targeting the 80 external to 80 internal and 443 external to 443 internal.
2 - Install the Nginx Proxy Manager Addon and start
3 - Create a host as it says in the manual and I created a creation then SSL, for example, an internal error and the log below

[5/6/2020] [2:50:27 PM] [Nginx ] › ℹ info Reloading Nginx
[5/6/2020] [2:50:27 PM] [SSL ] › ℹ info Requesting Let'sEncrypt certificates for Cert #10: xyz.duckdns.org
[5/6/2020] [2:50:50 PM] [Nginx ] › ℹ info Reloading Nginx
[5/6/2020] [2:50:50 PM] [Express ] › ⚠ warning Command failed: /usr/bin/certbot certonly --non-interactive --config "/etc/letsencrypt.ini" --cert-name "npm-10" --agree-tos --email "xyz@gmail.com" --preferred-challenges "dns,http" --webroot --domains "xyz.duckdns.org"
Saving debug log to /data/logs/letsencrypt/letsencrypt.log
Plugins selected: Authenticator webroot, Installer None
Obtaining a new certificate
Performing the following challenges:
http-01 challenge for xyz.duckdns.org
Using the webroot path /data/letsencrypt-acme-challenge for all unmatched domains.
Waiting for verification...
Challenge failed for domain xyz.duckdns.org
http-01 challenge for xyz.duckdns.org
Cleaning up challenges
Some challenges have failed.

There is any solution ?

**Comment by MapGuy11** [comment]
Yes! Look At The Bottom of this page With My Username And There is a solution! Your ISP is probably blocking 80 and 443. 
https://github.com/jc21/nginx-proxy-manager/issues/378

**Comment by flipepiloto** [comment]
> Sim! Veja a parte inferior desta página com meu nome de usuário e existe uma solução! Seu ISP provavelmente está bloqueando 80 e 443.
> [jc21 / nginx-proxy-manager # 378](https://github.com/jc21/nginx-proxy-manager/issues/378)

241/5000
If I'm right, you used https://www.sslforfree.com/ to create a certificate for your xyz.duckdns.org.
How did you associate this certificate with Nginx Proxy Manager?
I am a little layman on the subject, if you can help me, I appreciate it.

**Comment by MapGuy11** [comment]
Ok!
1. Go To sslforfree.com
2. put in *.xyz.duckdns.org
3. Click Manually Verify Domain
4. Copy The Value It Gives You 
5. https://www.duckdns.org/update?domains=xyz.duckdns.org&token=onduckdns.org&txt=fromsslforfree
5a. Use This to upload to DuckDNS
6. Go Back To SSLFORFREE and Click Download SSL Certificate
7. If Evrything Works you will get a .zip file
8. Go TO NGINX Proxy Manager
9. Go to SSL
10. Click Add Certificate at top Right Of Screen Then Click Custom
11. Name Is *.xyz.duckdns.org
12. Certificate Key Is In .zip file under private.key
13. Certificate is In .zip file under certificate.crt
14. Then Save!

**Comment by MapGuy11** [comment]
I hope this helps!

**Comment by TWhidden** [comment]
Still, even with your solution using a 3rd party (which is also a lets-encrypt cert, which is good for 3 months if I recall correctly), the whole point of this cool add-on is to handle the renewals and registration for you.  There are other cool lets-encrypt docker containers out there (and do the DNS challenge, but none are arm64 based that I could fine), but being that this is a slick little HA addon, would be nice not to have to deal with the other players and then mounting of their output to this input which still wouldn't be automated.  Considering a fork, dev implement, and pull request on this, but just too many other projects going on.  So this thread is more for lobbying to add in DNS.  (_and if the author of this addin is reading -- I use Google Cloud DNS services :)_ )  Would be more than willing to try a beta image if you ever get too it.

**Comment by MapGuy11** [comment]
Yes, I know that but the features It gives I am fine with updating one cert every 3 months not to hard.

**Comment by flipepiloto** [comment]
> Ok!
> 
> 1. Go To sslforfree.com
> 2. put in *.xyz.duckdns.org
> 3. Click Manually Verify Domain
> 4. Copy The Value It Gives You
> 5. https://www.duckdns.org/update?domains=xyz.duckdns.org&token=onduckdns.org&txt=fromsslforfree
>    5a. Use This to upload to DuckDNS
> 6. Go Back To SSLFORFREE and Click Download SSL Certificate
> 7. If Evrything Works you will get a .zip file
> 8. Go TO NGINX Proxy Manager
> 9. Go to SSL
> 10. Click Add Certificate at top Right Of Screen Then Click Custom
> 11. Name Is *.xyz.duckdns.org
> 12. Certificate Key Is In .zip file under private.key
> 13. Certificate is In .zip file under certificate.crt
> 14. Then Save!

HI MapGuy11,

First of all, I would like to thank you for your help.
I am a layman in these matters and I was in doubt in the 3rd step.
Which of the options below should I click ??
Thanks again for your help.
![choices](https://user-images.githubusercontent.com/63123077/81301686-c16a3280-904f-11ea-8650-18d2436b5db9.jpg)

**Comment by MapGuy11** [comment]
You forgot to put the *.xyz.duckdns.org redo then steps will make sense

**Comment by MapGuy11** [comment]
I am creating a video to help you. It will be published soon.

**Comment by flipepiloto** [comment]
I'm waiting anxiously!!

**Comment by MapGuy11** [comment]
Almost Uploaded Sending Link Soon!

**Comment by MapGuy11** [comment]
HERE IS THE VIDEO!!! https://youtu.be/KAkl1mDbcxM

**Comment by bieniu** [comment]
@flipepiloto You can use Caddy Server. It's a www/revers proxy server with automatic generation of SSL certs from Let's Encrypt with DNS challenge.

**Comment by flipepiloto** [comment]
> HERE IS THE VIDEO!!! https://youtu.be/KAkl1mDbcxM
> 
> > <img alt="" width="48" src="https://camo.githubusercontent.com/1a3f9889f25b820876acacd18e418f207789ecff/68747470733a2f2f692e7974696d672e636f6d2f76692f4b416b6c316d446263784d2f687164656661756c742e6a7067"><img alt="" height="14" src="/yts/img/favicon_144-vfliLAfaB.png"> YouTube**[Guide To Add SSL On NGINX Proxy Manager](https://www.youtube.com/watch?v=KAkl1mDbcxM)**For People Who Have 80 and 443 blocked by ISP most people around the world!

Hi MapGuy11,

Tks a lot for your help.
Follow your video i got the same result as you!!
But when i try to access my domains https://pilotofilipe.duckdns.org:8123 i get conection refused.
I need this to use Google Assistant on my Home Assistant.

Could you help me ??
![Sem título](https://user-images.githubusercontent.com/63123077/81331531-1d48b180-9078-11ea-9dc9-a54c311776b7.jpg)

**Comment by frenck** [comment]
Duplicate of #69
